Service/SchedulerService.php

```diff
@@ -0,0 +1,186 @@
+    public function enable      () { return $this->commandAction('enable'); }
+    public function setOndemand () { return $this->commandAction( ScheduledCommand::MODE_ONDEMAND); }
+    public function setAuto     () { return $this->commandAction( ScheduledCommand::MODE_AUTO); }
+    
+    /**
+     * Command Statuses
+     */
+    public function isFailing  () { return $this->commandStatus('failing'); }
+    public function isRunning  () { return $this->commandStatus('running'); }
+    public function isStopped  () { return $this->commandStatus('stopped'); }
+    public function isDisabled () { return $this->commandStatus('disabled'); }
+    public function isEnabled  () { return $this->commandStatus('enabled'); }
+    public function isOndemand () { return $this->commandStatus( ScheduledCommand::MODE_ONDEMAND); }
+    public function istAuto    () { return $this->commandStatus( ScheduledCommand::MODE_AUTO); }
+    
+    /**
+     *
+     * @return ScheduledCommand
+     * @throws \ErrorException
+     */
+    private function getCommand ( ) {
+        if ( $this->command) {
+            return $this->command;
+        }
+        
+        if ( !$this->commandName ){
+            throw new \ErrorException('Missing Command Name.');
+        }
+        
+        $cmd = $this->doctrine->getRepository(ScheduledCommand::class)->findOneBy([
+                    'name' => $this->commandName
+                ]);
+        
+        if ( $cmd instanceof ScheduledCommand) {
+            return $cmd;
+        }
+        
+        throw new \ErrorException('Command Not Found.');
+    }
+
+
+    /**
+     * Schedule command to be executed in the next execution cycle
+     *
+     * @param string $action
+     * @return bool
+     */
+    private function commandAction ( string $action ){
+        try {            
+            $cmd = $this->getCommand();
+
+            switch ( $action ){
+                case 'run': $cmd->setExecuteImmediately(true); break;
+                case 'stop': $cmd->setExecuteImmediately(false); break;
+                case 'disable': $cmd->setDisabled(true); break;
+                case 'enable': $cmd->setDisabled(false); break;
+                case ScheduledCommand::MODE_ONDEMAND: $cmd->setExecutionMode( ScheduledCommand::MODE_ONDEMAND); break;
+                case ScheduledCommand::MODE_AUTO: 
+                    if ( CronExpressionLib::isValidExpression( $cmd->getCronExpression()) ) {
+                        $cmd->setExecutionMode( ScheduledCommand::MODE_AUTO); 
+                    } else {
+                        throw new \InvalidArgumentException('Invalid Cron Expression.');
+                    }
+                    break;
+            }
+            $this->doctrine->getManager()->persist($cmd);
+            $this->doctrine->getManager()->flush($cmd);
+
+            return true;
+        } catch (\Exception $ex) {// TODO: Improve error handling            
```

Comment:
TODO: Improve error handling
I'd feel good if it was changed to an @FUTURE and some log output was used rather than silently ignoring the exception.